server/src/test/java/org/opensearch/gateway/remote/RemoteStateTestUtil.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway.remote;
+
+import org.opensearch.core.action.ActionListener;
+
+public class RemoteStateTestUtil {
```

Comment:
No need for this class or the additional file. Just put `TestCapturingListener` as a private static nested class in RemoteGlobalMetadataManagerTests (the only place that it is used). If you want to re-use TestCapturingListener more widely later you can consider refactoring at that point. And if that did happen I wouldn't nest it inside "RemoteStateTestUtil" because there is nothing remote state-specific about this.

Reply:
Actually, I am planning to use this in more tests in `gateway.remote` and have created another PR https://github.com/opensearch-project/OpenSearch/pull/14427.
Where can we put it if we want to keep it generic? Does it make sense to keep it in [test/framework/src/main/java/org/opensearch/common/util](https://github.com/opensearch-project/OpenSearch/tree/main/test/framework/src/main/java/org/opensearch/common/util)?

Reply:
Maybe just put it here without the outer RemoteStateTestUtil class then? Probably fine to move it somewhere in the test framework lib too.

Reply:
Moved the class to test framework as discussed.